rego/rego_wasmtarget_test.go

```diff
@@ -149,12 +149,19 @@ allow {
 }`
 
 	// Also a host function, but uses context.Context for cancellation.
+	// Use a different policy for the wasm run to avoid http.send's internal client caching mechanism
```

Comment:
Why do we need to avoid that mechanism here? I'm not sure I understand why the client caching should make these tests fail... 🤔 

How does the method used in a http.send affect whether the underlying `http.Client` is coming from the cache or not..? 💭

Reply:
These tests have a context timeout set to 1 sec. The expectation seems to be that the context gets cancelled and the test then checks the cancellation error. With the change in `http.send`, the wasm http send test case completes within a second and hence the test fails. This case fetches the client from the cache instead of creating a new one. I've tried to address this here.

Reply:
We don't cache server responses. I think the issue is we cache the request but don't update its context when we fetch it from the cache. I'll update that.

Reply:
Updated.